internal/rpc/server_test.go

```diff
@@ -32,55 +32,31 @@ import (
 
 func TestStartStopServer(t *testing.T) {
 	assert := assert.New(t)
-	grpcLh := MustListen()
-	httpLh := MustListen()
+	grpcL := MustListen()
+	httpL := MustListen()
 	ff := &shellTesting.FakeFrontend{}
 
-	params := NewServerParamsFromListeners(grpcLh, httpLh)
+	params := NewServerParamsFromListeners(grpcL, httpL)
 	params.AddHandleFunc(func(s *grpc.Server) {
 		pb.RegisterFrontendServiceServer(s, ff)
 	}, pb.RegisterFrontendServiceHandlerFromEndpoint)
 	s := &Server{}
 	defer s.Stop()
 
-	waitForStart, err := s.Start(params)
+	err := s.Start(params)
 	assert.Nil(err)
-	waitForStart()
```

Comment:
what is the recommended pattern to check if a server is started or not? I think `waitForStart` is the best effort to avoid test flakyness at here.

Reply:
The trick is to know that waitForStart never did anything in the first place (see pr description.)

What really matters is that the listeners are started - even if the server's aren't accepting requests off of the listener yet, connections won't fail due to being rejected.  Instead they'll just wait until they are accepted (or time out, in which case other problems).

Reply:
I agree, I think we should add a [WaitForReady](https://godoc.org/google.golang.org/grpc#WaitForReady) call option by default to our internal end to end tests.

Reply:
Possibly, though

> WaitForReady configures the action to take when an RPC is attempted on broken connections or unreachable servers.

This won't be the case for the in memory tests.  The server will be reachable (because the listener is started.)